### PR TITLE
Feature/dsnd 514 deserialise

### PR DESCRIPTION
**Pull request by vtondepu**
DSND-514: Changes made to deserialise the avro records from kafka.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessor.java [hunk 1]
```diff
@@ -35,16 +42,20 @@ public ChargesDeltaProcessor(ChargesDeltaProducer deltaProducer,
...
-            final ChsDelta payload = chsDelta.getPayload();
 
             ObjectMapper mapper = new ObjectMapper();
-            //TODO: To use the ChargesDelta
-            //InsolvencyDelta insolvencyDelta = mapper.readValue(payload.getData(),
-            //        InsolvencyDelta.class);
-
-            //transformer.transform(insolvencyDelta);
+            ChargesDelta chargesDelta = mapper.readValue(payload.getData(), ChargesDelta.class);
+            logger.trace(String.format("ChargesDelta extracted "
```
Comment (zahidm3) [inline src/main/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessor.java:51]: I think you need to mention the Jira Id on trace statements for QA team

File: src/test/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessorTest.java [hunk 2]
```diff
@@ -67,36 +84,47 @@ void setUp() {
...
+        charge.setCompanyNumber("01099198");
+        charge.setDeltaAt("20211029142043360560");
+        charge.setId("3387778");
+
+        List<Person> personList = new ArrayList<>();
+        Person person = new Person();
+        person.setPerson("LOMBARD NORTH CENTRAL PLC");
+        personList.add(person);
+        charge.setPersonsEntitled(personList);
+
```
Comment (zahidm3) [inline src/test/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessorTest.java:101]: Maybe move this to ChargesDeltaClass in model package to use it in different tests?

File: src/main/java/uk/gov/companieshouse/charges/delta/serialization/ChsDeltaDeserializer.java [hunk 3]
```diff
@@ -8,19 +8,34 @@
...
+
+    @Autowired
+    public ChsDeltaDeserializer(Logger logger) {
+        this.logger = logger;
+    }
+
     @Override
     public ChsDelta deserialize(String topic, byte[] data) {
         try {
+            logger.trace(String.format("Message picked up from topic with data: %s",
```
Comment (gseregni-ch) [inline src/main/java/uk/gov/companieshouse/charges/delta/serialization/ChsDeltaDeserializer.java:29]: I would like to suggest to change this to:
logger.trace(String.format("DSND-514: Message picked up from topic with data: %s",
                    new String(data)));

**Review by gseregni-ch — APPROVED** [review]

**Comment by vtondepu** [comment]
Addressed all the PR comments.

**Review by zahidm3 — APPROVED** [review]

**Review by gseregni-ch — APPROVED** [review]

**Comment by vtondepu** [comment]
2 approvals acquired (@zahidm3  and  @gseregni-ch approved ) hence merging the PR.

**Pull request merged**